### PR TITLE
upcoming: [M3-8019] – Add Encrypted/Not Encrypted status to Linode Detail summary header

### PR DESCRIPTION
**Pull request by dwiley-akamai**
## Description 📝
Add Encrypted/Not Encrypted status to Linode Detail summary header.

## Changes  🔄
- Add `lke_cluster_id` to Linode interface, update mocks and several test files accordingly
- Add `Encrypted` and `Not Encrypted` status to Linode Detail summary header

## Target release date 🗓️
6/10

## Preview 📷
<details>
<summary>Screenshots</summary>

Encrypted:
Screenshot 2024-05-31 at 2 55 17 PM

Unencrypted & is LKE Linode:
Screenshot 2024-05-31 at 2 57 22 PM

Unencrypted & is standard Linode:
Screenshot 2024-05-31 at 2 58 22 PM

</details>

## How to test 🧪
To test each of the three cases outlined by the screenshots, it'll be easiest to use the MSW and make adjustments to these lines:

https://github.com/linode/manager/blob/0e2d367d192f9a28b92b5b37ebd7f566c345ef5c/packages/manager/src/mocks/serverHandlers.ts#L768-L778

```typescript
      linodeFactory.build({
        backups: { enabled: false },
        disk_encryption: 'disabled', // change to 'enabled' to see encrypted state
        id,
        label: 'Gecko Edge Test',
        lke_cluster_id: null, // change to a number to see the LKE Linode state
        region: 'us-edge-1',
      })
```

### Prerequisites
1. Have the linode_disk_encryption tag on your account
2. Point at the alpha/dev environment
3. Have the Linode Disk Encryption (LDE) flag in our dev tool toggled on

### Verification steps
Without the tag and/or with the LDE flag in our dev tool toggled off, you should not see any LDE-related things in the UI. Otherwise,

1. Navigate to a Linode Details page
2. Confirm the displayed encryption status matches what the `disk_encryption` field in the linode detail network request indicates (`enabled` --> `Encrypted`, `disabled` --> `Not Encrypted`)
3. If "Not Encrypted", standard Linodes should have a tooltip that reads: "Use Rebuild to enable or disable disk encryption." LKE Linodes should have a tooltip that reads: "To enable disk encryption, delete the node pool and create a new node pool. New node pools are always encrypted." 
4. Ensure the header looks clean as you shrink and expand the viewport (made some styling adjustments to accommodate the encrypted status)

## As an Author I have considered 🤔
- [X] 👀 Doing a self review
- [X] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [X] 🤏 Splitting feature into small PRs
- [X] ➕ Adding a changeset
- [X] 🧪 Providing/Improving test coverage
- [X] 🔐 Removing all sensitive information from the code and PR description
- [X] 🚩 Using a feature flag to protect the release
- [X] 👣 Providing comprehensive reproduction steps
- [X] 📑 Providing or updating our documentation
- [X] 🕛 Scheduling a pair reviewing session
- [X] 📱 Providing mobile support
- [X] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/DiskEncryption/constants.tsx [hunk 1]
```diff
@@ -22,3 +22,9 @@ export const DISK_ENCRYPTION_BACKUPS_CAVEAT_COPY =
 
 export const DISK_ENCRYPTION_NODE_POOL_GUIDANCE_COPY =
   'To enable disk encryption, delete the node pool and create a new node pool. New node pools are always encrypted.';
+
+export const UNENCRYPTED_STANDARD_LINODE_GUIDANCE_COPY =
+  'Use Rebuild to enable or disable disk encryption.';
```
Comment (mjac0bs) [inline packages/manager/src/components/DiskEncryption/constants.tsx:27]: This copy feels a little strange when it begins with the two verbs next to each other. Would something like `Rebuild your Linode to enable or disable disk encryption.` be better here?

**Review by mjac0bs — COMMENTED** [review]
- Confirmed that with the LDE flag in our dev tool toggled off, I do not see any LDE-related things in the UI. 
- Confirmed the displayed encryption status matches what the disk_encryption field in the linode detail network request indicates (enabled --> Encrypted, disabled --> Not Encrypted).
- Confirmed the tooltip matches the screenshots in the Preview section for standard vs LKE linodes.
- Confirmed the encrypted status and tooltip look good upon screen resizing.
- Confirmed no styling regressions to the encryption status on the LKE details page.

Thanks for the test coverage too! Left a few minor comments.

**Review by hkhalil-akamai — APPROVED** [review]
Verified detail header correctly reflects `disk_encryption` value and layout scales with the window as expected.

Approved pending @mjac0bs' comments regarding changelog and copy wording.

File: packages/manager/.changeset/pr-10537-added-1717182633416.md
[the remark this answers is not in the document]
Comment (dwiley-akamai) [reply packages/manager/.changeset/pr-10537-added-1717182633416.md:1]: It's an upcoming feature (accidentally misclassified the changeset initially).

File: packages/api-v4/.changeset/pr-10537-changed-1717182754934.md
[the remark this answers is not in the document]
Comment (dwiley-akamai) [reply packages/api-v4/.changeset/pr-10537-changed-1717182754934.md:1]: From my perspective, for the `api-v4` package, "Added" indicates a newly-created type or interface, whereas "Changed" would include existing types and interfaces being expanded with new properties. Open to changing this based on additional thoughts, but it's probably a good Cafe item so a convention can be set.

File: packages/manager/src/components/DiskEncryption/constants.tsx [hunk 2]
```diff
@@ -22,3 +22,9 @@ export const DISK_ENCRYPTION_BACKUPS_CAVEAT_COPY =
...
+
+export const UNENCRYPTED_LKE_LINODE_GUIDANCE_COPY =
```
[the remark this answers is not in the document]
Comment (dwiley-akamai) [reply packages/manager/src/components/DiskEncryption/constants.tsx]: Yeah, we shouldn't duplicate it. `UNENCRYPTED_LKE_LINODE_GUIDANCE_COPY` was initially different copy, but the wireframes got updated and we ended up reusing the same copy but I didn't get rid of the variable.

To preserve the contextual clarity in `LinodeEntityDetailBody.tsx` I imported `DISK_ENCRYPTION_NODE_POOL_GUIDANCE_COPY as UNENCRYPTED_LKE_LINODE_GUIDANCE_COPY` in the most recent commit, is that agreeable or do you think a renaming is still needed?

[the remark this answers is not in the document]
Comment (mjac0bs) [reply packages/manager/src/components/DiskEncryption/constants.tsx]: Thanks! I think how you addressed it is fine.

File: packages/api-v4/.changeset/pr-10537-changed-1717182754934.md
[the remark this answers is not in the document]
Comment (mjac0bs) [reply packages/api-v4/.changeset/pr-10537-changed-1717182754934.md:1]: Your perspective makes sense - I am curious to hear what other members of the team think fits best here. Totally possible that I'm just thinking about it differently.

**Review by mjac0bs — APPROVED** [review]
Approving, pending we follow up on possible copy update (pending UX feedback) in this ticket or a future one. 🚀

File: packages/manager/src/components/DiskEncryption/constants.tsx (hunk 1 above)
Comment (dwiley-akamai) [reply packages/manager/src/components/DiskEncryption/constants.tsx]: Revised to `Rebuild this Linode to enable or disable disk encryption.` after revisiting the copy with UX 🚀

File: packages/api-v4/.changeset/pr-10537-changed-1717182754934.md
[the remark this answers is not in the document]
Comment (mjac0bs) [reply packages/api-v4/.changeset/pr-10537-changed-1717182754934.md:1]: (In support of leaving this changeset as is, I can add a comment to our next changelog doc when the team reviews pre-release just to clarify our convention, and we won't need to dedicate any cafe time.)

**Pull request merged**